src/Meshes/interval.jl

```diff
@@ -318,6 +319,91 @@ function IntervalMesh(
     IntervalMesh(domain, CT.(faces))
 end
 
+"""
+    HyperbolicTangentStretching(dz_surface::FT)
+
+Apply a hyperbolic tangent stretching to the domain when constructing elements.
+`dz_surface` is the target element grid spacing at the surface. In typical atmosphere
+configuration, it is the grid spacing at the bottom of the
+vertical column domain (m). On the other hand, for typical land configurations,
+it is the grid spacing at the top of the vertical column domain.
+
+For an interval ``[z_0,z_1]``, this makes the elements uniformally spaced in
+``\\zeta``, where
+```math
+\\eta = 1 - \\frac{tanh[\\gamma(1-\\zeta)]}{tanh(\\gamma)},
+```
+where ``\\eta = \\frac{z - z_0}{z_1-z_0}``. The stretching parameter ``\\gamma``
+is chosen to achieve a given resolution `dz_surface` at the surface. 
+
+Then, the user can define a stretched mesh via
+
+    ClimaCore.Meshes.IntervalMesh(interval_domain, HyperbolicTangentStretching(dz_surface); nelems::Int, reverse_mode)
+
+`reverse_mode` is default to false for atmosphere configurations. For land configurations, 
+use `reverse_mode` = `true`.
+"""
+struct HyperbolicTangentStretching{FT} <: StretchingRule
+    dz_surface::FT
+end
+
+function IntervalMesh(
+    domain::IntervalDomain{CT},
+    stretch::HyperbolicTangentStretching{FT};
+    nelems::Int,
+    FT_solve = Float64,
+    tol = 1e-5,
+    reverse_mode::Bool = false,
+) where {CT <: Geometry.Abstract1DPoint{FT}} where {FT}
+    if nelems ≤ 1
+        throw(ArgumentError("`nelems` must be ≥ 2"))
+    end
+
+    dz_surface = FT_solve(stretch.dz_surface)
+
+    # bottom coord height value, always min, for both atmos and land, since z-axis does not change
```

Comment:
I'm confused by this comment. If this is bottom, for land, why would it be the min? Perhaps you can clarify what this comment is supposed to mean.

Reply:
I copied this comment from the other stretching method. But I think it is because for land z is also positive upward (?), so the bottom z has the minimum (most negative) value.

Reply:
Maybe just explain it better then.